### PR TITLE
Moved privsec from W3C to Core Linter Rules

### LINKED ISSUE #2223: Move priv-sec linter rule to core
The Geonovum folks want to use this module too, so we should make it a core module.

### PR DESCRIPTION
**Pull request by Swapnilr1**
Fixes #2223 

BTW, this is not refactoring or a fix or a feature. Not sure how to appropriately title this PR.

### CODE REVIEW

File: src/core/defaults.js [hunk 1]
```diff
@@ -27,6 +29,7 @@ export const coreDefaults = {
     "local-refs-exist": true,
     "check-internal-slots": false,
     "check-charset": false,
+    "privsec-section": true,
```
Comment (marcoscaceres) [inline src/core/defaults.js]: False by default in core... but `true` in the W3C profile.

Comment (marcoscaceres) [inline src/core/defaults.js]:
```suggestion
    "privsec-section": false,
```

File: src/w3c/defaults.js [hunk 2]
```diff
@@ -5,14 +5,11 @@ export const name = "w3c/defaults";
 import { coreDefaults } from "../core/defaults";
 import { definitionMap } from "../core/dfn-map";
 import linter from "../core/linter";
-import { rule as privsecSectionRule } from "./linter-rules/privsec-section";
+import { rule as privsecSectionRule } from "../core/linter-rules/privsec-section";
 
 linter.register(privsecSectionRule);
 
 const w3cDefaults = {
-  lint: {
```
Comment (marcoscaceres) [inline src/w3c/defaults.js]: Bring this one back :)

Comment (Swapnilr1) [reply src/w3c/defaults.js]: Sure. Working on it.

File: tests/spec/core/linter-rules/privsec-section-spec.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ describe("W3C Linter Rule - 'privsec-section'", () => {
   let rule;
   beforeAll(async () => {
     rule = await new Promise(resolve => {
-      require([`w3c/linter-rules/${ruleName}`], ({ rule }) => resolve(rule));
+      require([`core/linter-rules/${ruleName}`], ({ rule }) => resolve(rule));
```
Comment (marcoscaceres) [inline tests/spec/core/linter-rules/privsec-section-spec.js]: Probably needs updating based on what @saschanaz just merged.

**Pull request merged**